Gems/Atom/RHI/Code/Include/Atom/RHI/ObjectCollector.h

```diff
@@ -172,6 +179,38 @@ namespace AZ
             {
                 m_pendingGarbage.push_back({ AZStd::move(m_pendingObjects), m_currentIteration });
             }
+
+            if (m_pendingNotifies.size())
+            {
+                if (m_pendingGarbage.size())
+                {
```

Comment:
Usually, it is better to check for `empty` state than query the size

Reply:
+1